### DISCUSSION TITLE
Where to find SvnClientAdapter related artifacts to include in a pom file?

### DISCUSSION THREAD

**Discussion by Steelhammer11** [subclipse/svnclientadapter#41] [General]
Hi
I am using the SvnClientAdapter for accessing an SVN repository from a java application. I was looking at the SvnClientAdapter  sample code in GitHub (https://github.com/subclipse/svnclientadapter/tree/master/samples), and in that sample code, in the pom file, there is a reference to a repository where the dependency artifacts are located, i.e.:

`<repositories>`
       `_<repository>`
            `__<id>bintray-subclipse-maven</id>`
            `__<name>bintray</name>`
            ` __<url>http://dl.bintray.com/subclipse/maven</url>`
        `_</repository>`
    `</repositories>`

I can manually find the artifacts using the referenced URL through a web browser, but not through Maven having the above in my pom file. Is it a valid Maven repository? Or are the SvnClientAdapter artifacts supposed to be found in another location? I was able to manually download the jars and include them in my pom as local system dependencies but I would prefer have them fetched from a Maven repo.

I can see that this particular code has not been touched for 4 years, but other parts of the code seem to at least have been touched 9 months ago updating it to SVN 1.14.0. However the latest artifact found in the repository mentioned above is 1.12.0. Is the artifact repository still maintained or should it be considered deprecated?

**Comment by markphip** [comment]
> I can manually find the artifacts using the referenced URL through a web browser, but not through Maven having the above in my pom file. Is it a valid Maven repository? 

It should be. It is just a Bintray feature and it was created as a Maven repository.

Go here, and click the Set Me Up button.  https://bintray.com/subclipse/maven/svnclientadapter

There is more to do than just update your pom.xml.

> I can see that this particular code has not been touched for 4 years, but other parts of the code seem to at least have been touched 9 months ago updating it to SVN 1.14.0. However the latest artifact found in the repository mentioned above is 1.12.0. Is the artifact repository still maintained or should it be considered deprecated?

Honestly, it has never really been maintained, it is just something we use in Subclipse.  I never setup any build automation for this repository and guess I never published a 1.14. I recall all I did is relax the version checks so that the same JAR can work with SVN 1.12-1.14.

Mark

**Reply by Steelhammer11** [reply]
Ok thanks. So it sounds like I should use another API for accessing SVN? The thing is that I had been working with a plugin to Eclipse that needed to access SVN, and with Subclipse installed I could use features from that from my code. For other custom things I wanted to do I used the SvnClientAdapter with JavaHL directly.

Then I started some other work with a standalone java app that also needed to access SVN, and then I turned to the SvnClientAdapter  and JavaHL that I "knew", but then I needed to find those artifacts somewhere. Hence the questions in my previous post. But if it's not meant to be used outside Subclipse, I guess I should be using something else instead.

**Reply by markphip** [reply]
I think the people that originally created and worked on svnClientAdapter intended it for what it is, but over time the only real user of it was Subclipse and all we cared about was the JavaHL support. So the command line adapter has not been maintained in over 10 years and the SVNKit adapter has always just relied on the fact that SVNKit implements JavaHL.

Anyway, the point being svnClientAdapter only gets changed when Subclipse needs something and honestly Subclipse is not really maintained much any more either.

If you are working with an Eclipse plugin then it seems like you should not be using Maven anyway. I would think your plugin would have a dependency on the svnclientadapter plugin and/or the subclipse core plugin and you would gain access to the libraries via the plugin dependencies basically the same way that Subclipse does.  It kind of depends on what your plugin does and how much value is added by going this way as opposed to just including SVNKit or JavaHL in your plugin directly.

**Reply by markphip** [reply]
Going back to the original question though, I think the Maven repository is likely fine and you just have not configured your Maven profile to use it.

**Reply by Okst1988** [reply]
Hey Mark,

Are these repos still active?

https://maven.pkg.github.com/subclipse/svnclientadapter
https://maven.pkg.github.com/subclipse/javahl

I downloaded the project and tried to build it, but when downloading dependencies from the https://maven.pkg.github.com/subclipse/javahl, I get 401 errors. Should they be removed from the pom files?

**Comment by Steelhammer11** [comment]
Yes, as I said, when doing the plugin it worked fine using dependencies to Subclipse. It was when I wanted to use similar code from a standalone java app, built with Maven, I got issues trying to fetch the jars as dependencies in the pom.

Anyway, thanks for your answers. I'll have a look at the "set me up" button.

**Comment by markphip** [comment]
I have not done anything with them but they have never really been used by
anyone and none of this is really what one could call maintained.

That said, I have this vague recollection that the GitHub Maven
repositories do not support anonymous access so you have to authenticate to
use them?

https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry

401 is an authentication challenge so that lines up.

Mark

**Reply by Okst1988** [reply]
Thanks for responding so quickly! Yeah, that would make sense for the 401 error. Is there any reason the dependencies are not deployed in maven central?

**Comment by markphip** [comment]
We just never used Maven ourselves back when we worked on this
